pkg/upgrade/upgrade.go

```diff
@@ -216,37 +216,40 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform) error
 		return nil
 	}
 
-	// If KfDef CRD is not found, we see it as a cluster not pre-installed v1 operator	// Check if kfdef are deployed
-	kfdefCrd := &apiextv1.CustomResourceDefinition{}
-	err := cli.Get(context.TODO(), client.ObjectKey{Name: "kfdefs.kfdef.apps.kubeflow.org"}, kfdefCrd)
-	if err != nil {
-		if apierrs.IsNotFound(err) {
-			// If no Crd found, return, since its a new Installation
-			return nil
-		} else {
-			return fmt.Errorf("error retrieving kfdef CRD : %v", err)
+	if platform == deploy.SelfManagedRhods {
+		// If KfDef CRD is not found, we see it as a cluster not pre-installed v1 operator	// Check if kfdef are deployed
```

Comment:
I wonder if extracting `KfDef` logic to its own `func` wouldn't improve the readability of this code.